### PR TITLE
Added optimizations section: Code Split and Code Split Bundles

### PR DESCRIPTION
**Pull request by Elrhino**

### CODE REVIEW

File: site/src/main/markdown/gwtp_v2/advanced/optimizations.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Optimizations
+When developing a web application, the JavaScript part tends to become more heavy as it progress and this put significant load on the application startup. *Code Splitting* allows the JavaScript code to be split into multiple files that will be loaded depending on the application section being used. Separating your GWTP application into logical and independent sections will provide natural points for code splitting. See [GWT Code Splitting](http://www.gwtproject.org/doc/latest/DevGuideCodeSplitting.html) for more details.
```
Comment (Pacane) [inline site/src/main/markdown/gwtp_v2/advanced/optimizations.md]: as it progresses
and this puts

File: site/src/main/markdown/gwtp_v2/advanced/optimizations.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Here is an example of Code Split Bundles using manually set ProviderBundles.
+
+```java
+@ProxyCodeSplitBundle(bundleClass = MyPresenterBundle.class, id = MyPresenterBundle.ID_Object1)
+interface MyProxy extends ProxyPlace<Object1> {
+}
+```
+
+### Using generated ProviderBundles
+If you use GWTP's generation of [ApplicationController](http://arcbees.github.io/GWTP/javadoc/apidocs/com/gwtplatform/mvp/client/ApplicationController.html), all bundles will be automatically generated for you, all that you need are string identifiers for each unique bundle. The best way to keep your bundles in order is to create an interface that identifies them.
```
Comment (Pacane) [inline site/src/main/markdown/gwtp_v2/advanced/optimizations.md]: reformulate this

**Comment by Elrhino** [comment]
@Pacane - I fixed your comments and added a better description for generated Provider Bundles.

**Comment by Pacane** [comment]
lgtm

File: site/src/main/markdown/gwtp_v2/advanced/optimizations.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+```
+
+The JavaScript code will now be compiled into a new JavaScript file and be loaded from the application server only when needed.
+
+## Code Split Bundles
+*Code Split Bundles* allow you to regroup compiled JavaScript files into a bundle separated from other JavaScript files. This is useful when multiple Presenters share the same code base (i.e. Nested Presenters). To use Code Split Bundles, you need to set up your own implementation of a [ProviderBundle](http://arcbees.github.io/GWTP/javadoc/apidocs/com/gwtplatform/common/client/ProviderBundle.html). ProviderBundles can be created manually or generated by GWTP.
+
+### Using manually set ProviderBundles
+For the manual approach, the `ProxyCodeSplitBundle.bundleClass()` and `ProxyCodeSplitBundle.id()` need to be defined.
```
Comment (jclariviere) [inline site/src/main/markdown/gwtp_v2/advanced/optimizations.md]: and ProxyCodeSplitBundle.id() "methods" (missing word)

File: site/src/main/markdown/gwtp_v2/advanced/optimizations.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+Here is an example of Code Split Bundles using manually set ProviderBundles.
```
Comment (jclariviere) [inline site/src/main/markdown/gwtp_v2/advanced/optimizations.md]: Might be useful to provide an example of the implementation of `ProxyCodeSplitBundle`.

**Comment by Elrhino** [comment]
@jclariviere - I added an example of a ProviderBundle implementation.

**Comment by jclariviere** [comment]
LGTM

**Pull request merged**